### PR TITLE
[ONNX] Added initial support of DequantizeLinear-21

### PR DESCRIPTION
**Pull request by gkrivor**
### Details:
 - Added initial support of operation DequantizeLinear-21 from default opset

### Tickets:
 - 152784

### CODE REVIEW

File: src/frontends/onnx/frontend/src/op/dequantize_linear.cpp [hunk 1]
```diff
@@ -188,8 +193,83 @@ ov::OutputVector dequantize_linear(const ov::frontend::onnx::Node& node) {
...
+}  // namespace opset_19
+
+namespace opset_21 {
+ov::OutputVector dequantize_linear(const ov::frontend::onnx::Node& node) {
+    const ov::OutputVector inputs{node.get_ov_inputs()};
+
+    FRONT_END_GENERAL_CHECK(2 <= inputs.size() && inputs.size() <= 3,
+                            "The DequantizeLinear op expects 2 required and one optional "
+                            "input. Got: ",
+                            inputs.size());
```
Comment (rkazants) [inline src/frontends/onnx/frontend/src/op/dequantize_linear.cpp:210]: please create helper for this check. We have similar helper in PT FE and TF FE

File: src/frontends/onnx/frontend/src/op/dequantize_linear.cpp [hunk 2]
```diff
@@ -188,8 +193,83 @@ ov::OutputVector dequantize_linear(const ov::frontend::onnx::Node& node) {
...
+    ov::Output<ov::Node> scale = inputs[1];
+    const auto& scale_shape = scale.get_partial_shape();
+    ov::Output<ov::Node> zp;
+
+    // When no blocking dequantization is required - use regular DequantizeLinear
+    if (scale_shape.rank().is_static() && scale_shape.rank().get_length() <= 1) {
+        return ai_onnx::opset_13::dequantize_linear(node);
+    }
+
+    FRONT_END_GENERAL_CHECK(scale_shape.rank().get_length() == 2,
```
Comment (rkazants) [inline src/frontends/onnx/frontend/src/op/dequantize_linear.cpp:221]: at this point are we sure that `rank` is static? I think - no:)
Please check

File: src/frontends/onnx/frontend/src/op/dequantize_linear.cpp [hunk 3]
```diff
@@ -188,8 +193,83 @@ ov::OutputVector dequantize_linear(const ov::frontend::onnx::Node& node) {
...
+                            "DequantizeLinear cannot operate with more than 2D scales");
+    FRONT_END_GENERAL_CHECK(src_x.get_partial_shape().is_static(),
+                            "DequantizeLinear cannot operate with dynamic shapes of input X");
+
+    // Working
+    //    const auto& unsqueezed_axes = std::make_shared<v0::Constant>(ov::element::i64, Shape{1},
+    //    std::vector<int64_t>{2});
```
Comment (rkazants) [inline src/frontends/onnx/frontend/src/op/dequantize_linear.cpp:228]:
```suggestion
```

File: src/frontends/onnx/frontend/src/op/dequantize_linear.cpp [hunk 4]
```diff
@@ -188,8 +193,78 @@ ov::OutputVector dequantize_linear(const ov::frontend::onnx::Node& node) {
...
+    const auto block_size = static_cast<size_t>(node.get_attribute_value<int64_t>("block_size", 0));
+    const auto scale_type = scale.get_element_type();
+
+    FRONT_END_GENERAL_CHECK(axis == 0, "Axis != 0 isn't supported");
+    FRONT_END_GENERAL_CHECK(block_size > 0, "block_size must be greater than zero");
+    FRONT_END_GENERAL_CHECK(
+        src_x.get_shape()[0] % block_size == 0,
+        "DequantizeLinear doesn't support case when first dimension of X cannot be divided by block_size");
+
+    const auto& x = src_x.get_element_type() == scale_type ? src_x : std::make_shared<v1::ConvertLike>(src_x, scale);
```
Comment (rkazants) [inline src/frontends/onnx/frontend/src/op/dequantize_linear.cpp:244]: can you please simplify? head breaking expression:)

File: src/frontends/onnx/frontend/src/op/dequantize_linear.cpp [hunk 5]
```diff
@@ -188,8 +193,78 @@ ov::OutputVector dequantize_linear(const ov::frontend::onnx::Node& node) {
...
+
+    // When no blocking dequantization is required - use regular DequantizeLinear
+    if (scale_shape.rank().is_static() && scale_shape.rank().get_length() <= 1) {
+        return ai_onnx::opset_13::dequantize_linear(node);
+    }
+
+    FRONT_END_GENERAL_CHECK(scale_shape.rank().is_static(), "Rank of the input data tensor has to be known (static).");
+    FRONT_END_GENERAL_CHECK(scale_shape.rank().get_length() == 2,
+                            "DequantizeLinear cannot operate with more than 2D scales");
+    FRONT_END_GENERAL_CHECK(src_x.get_partial_shape().is_static(),
```
Comment (rkazants) [inline src/frontends/onnx/frontend/src/op/dequantize_linear.cpp:221]: really important limitation that can be often not met. We can think of how to support it via decomposition one more time, or request new op in the opset.

**Review by rkazants — APPROVED** [review]

File: src/frontends/onnx/frontend/src/op/dequantize_linear.cpp (hunk 4 above)
Comment (gkrivor) [reply src/frontends/onnx/frontend/src/op/dequantize_linear.cpp:244]: Ok, in a separate PR, CI is finally green

**Pull request merged**